chore: Add release notes for Mender 3.7.1 (C++ client) (#63)
mendersoftware/mender-docs-changelog: 20.Mender-client/docs.md

| Adam Duskett | 535 (0.5%) |
| Lars Erik Wik | 330 (0.3%) |
| Lukasz Finster | 138 (0.1%) |
| Josef Holzmayr | 114 (0.1%) |
| Alf-Rune Siqveland | 42 (0.0%) |

| Developers with the most lines removed | |
|---|---|
| Kristian Amlie | 19656 (31.9%) |
| Adam Duskett | 132 (0.2%) |
| Josef Holzmayr | 81 (0.1%) |
| Fabio Tranchitella | 1 (0.0%) |

| Developers with the most signoffs (total 6) | |
|---|---|
| Vratislav Podzimek | 3 (50.0%) |
| Ole Petter Orhagen | 2 (33.3%) |
| Kristian Amlie | 1 (16.7%) |

| Top changeset contributors by employer | |
|---|---|
| Northern.tech | 1001 (99.4%) |
| Tronel | 3 (0.3%) |
| aduskett@gmail.com | 2 (0.2%) |
| BlackBerry Limited | 1 (0.1%) |

| Top lines changed by employer | |
|---|---|
| Northern.tech | 117379 (99.4%) |
| aduskett@gmail.com | 535 (0.5%) |
| Tronel | 138 (0.1%) |
| BlackBerry Limited | 32 (0.0%) |

| Employers with the most signoffs (total 6) | |
|---|---|
| Northern.tech | 6 (100.0%) |

| Employers with the most hackers (total 16) | |
|---|---|
| Northern.tech | 13 (81.2%) |
| aduskett@gmail.com | 1 (6.2%) |
| Tronel | 1 (6.2%) |
| BlackBerry Limited | 1 (6.2%) |

### Changelogs

#### mender (4.0.0)

> New changes in mender since 3.5.2:

Reviewer: No mentions this is the "big rewrite project"?

Author: I can add here some note pointing to the mender-docs page that explains the upgrade and split. It depends on:
* https://github.com/mendersoftware/mender-docs/pull/2289

Author: I can add here some note pointing to the mender-docs page that explains the upgrade and split. It depends on:
* https://github.com/mendersoftware/mender-docs/pull/2289

Reviewer 2: Ah yes, I was planning to make a bigger note about it. If you want, I can take over this PR, and combine with my work in the other PR. Just assign it to me.

Author: Thank you :pray:

Reviewer 2: Mentioned the rewrite at the top level, with a link to the documentation for the migration guide.

##### Bug Fixes

* disabled implicit conversion form Error to bool
  ([MEN-6409](https://northerntech.atlassian.net/browse/MEN-6409))
* README: add build instructions for C++-client
* latest openssl reports "no such file" if the cert doesn't exist
* Rootfs scripts in the C++ client are checked in the same way as
  `Artifact` scripts, and hence, we are now moving it to support only `version 3`.
  While in the old `Golang` client both version `2` and `3` were supported.
  ([MEN-6671](https://northerntech.atlassian.net/browse/MEN-6671))
* The client no longer erronously commits a rootfs-image
  artifact after being restarted using `systemctl restart` in the
  `ArtifactReboot` state.
  ([MEN-6633](https://northerntech.atlassian.net/browse/MEN-6633))

##### Features

* build and test using the latest version of golang (1.21 today)

>   ([QA-614](https://northerntech.atlassian.net/browse/QA-614))

Reviewer: This doesn't make too much sense

Author: It doesn't make sense anymore.

It seems like [this change](https://github.com/mendersoftware/mender/commit/81f5e35172fc08db04120db8dd6669d76c8e9ebd) was marked as feature and we never ported it to 3.5.x. Hence it never got released it.

I will manually remove this entry.

| Adam Duskett | 535 (0.5%) |
| Lars Erik Wik | 330 (0.3%) |
| Lukasz Finster | 138 (0.1%) |
| Josef Holzmayr | 114 (0.1%) |
| Alf-Rune Siqveland | 42 (0.0%) |

| Developers with the most lines removed | |
|---|---|
| Kristian Amlie | 19656 (31.9%) |
| Adam Duskett | 132 (0.2%) |
| Josef Holzmayr | 81 (0.1%) |
| Fabio Tranchitella | 1 (0.0%) |

| Developers with the most signoffs (total 6) | |
|---|---|
| Vratislav Podzimek | 3 (50.0%) |
| Ole Petter Orhagen | 2 (33.3%) |
| Kristian Amlie | 1 (16.7%) |

| Top changeset contributors by employer | |
|---|---|
| Northern.tech | 1001 (99.4%) |
| Tronel | 3 (0.3%) |
| aduskett@gmail.com | 2 (0.2%) |
| BlackBerry Limited | 1 (0.1%) |

| Top lines changed by employer | |
|---|---|
| Northern.tech | 117379 (99.4%) |
| aduskett@gmail.com | 535 (0.5%) |
| Tronel | 138 (0.1%) |
| BlackBerry Limited | 32 (0.0%) |

| Employers with the most signoffs (total 6) | |
|---|---|
| Northern.tech | 6 (100.0%) |

| Employers with the most hackers (total 16) | |
|---|---|
| Northern.tech | 13 (81.2%) |
| aduskett@gmail.com | 1 (6.2%) |
| Tronel | 1 (6.2%) |
| BlackBerry Limited | 1 (6.2%) |

### Changelogs

#### mender (4.0.0)

> Mender client version 4.0.0 has been completely rewritten from scratch in C++. This comes with a
> number of changes which are important to pay attention to to ensure a successful migration. The

Author: ```suggestion
Mender client version 4.0.0 has been completely rewritten from scratch in C++. To ensure a successful migration, there are a number of changes which are important to pay attention to. The
```
